### PR TITLE
MOD-667 sorting activities refactored

### PR DESCRIPTION
**Pull request by umutyorulmaz**
### Summary
* Reduce function omitted
* Poh taken into consideration
* fetch minimized

### Type of Change
- [ ] New feature - Sorting
- [ ] Refactoring

### Gif

[d428bcb9-90bb-4f68-bf07-5af687b221ff.webm](https://github.com/modclub-app/modclub_src/assets/82664734/43a296b3-542d-4086-b843-23735d91620c)

### CODE REVIEW

File: src/modclub_assets/src/components/app/profile/Activity.tsx [hunk 1]
```diff
@@ -78,8 +142,28 @@ export default function Activity() {
...
+  // This useEffect will now also re-sort the activities whenever sortOrder changes
   useEffect(() => {
-    fetchActivity(currentFilter);
+    const sortAndSetActivities = async () => {
+      setLoading(true);
+
+      let activities;
+      if (modclub) {
+        activities =
+          currentFilter === "new"
```
[the remark this answers is not in the document]
Comment (umutyorulmaz) [reply src/modclub_assets/src/components/app/profile/Activity.tsx:153]: should i explain with a comment or define a constant like new_constant = "new" and then replace all "new"s with the constant?

[the remark this answers is not in the document]
Comment (umutyorulmaz) [reply src/modclub_assets/src/components/app/profile/Activity.tsx:153]: at line 41 this array: const filters = ["completed", "new"];

[the remark this answers is not in the document]
Comment (xwdeng) [reply src/modclub_assets/src/components/app/profile/Activity.tsx:152]: em, let's leverage typescript: modify line 36
```
type FilterType = 'new' | 'primary' | 'ghose';
const [currentFilter, setCurrentFilter] = useState<FilterType>('new');
```

File: src/modclub_assets/src/components/app/profile/Activity.tsx [hunk 2]
```diff
@@ -33,7 +33,12 @@ export default function Activity() {
     useState<boolean>(false);
   const [newEmail, setNewEmail] = useState<string>("");
   const [editEmail, setEditEmail] = useState<boolean>(false);
-  const [currentFilter, setCurrentFilter] = useState<string>("new");
+
+  type FilterType = "new" | "primary" | "ghost" | "completed";
```
Comment (xwdeng) [inline src/modclub_assets/src/components/app/profile/Activity.tsx:37]: nice

Comment (xwdeng) [inline src/modclub_assets/src/components/app/profile/Activity.tsx:37]:
```suggestion
  type FilterType = "new" | "completed";
```

File: src/modclub_assets/src/components/app/profile/Activity.tsx [hunk 3]
```diff
@@ -33,7 +33,12 @@ export default function Activity() {
...
+  const [currentFilter, setCurrentFilter] = useState<FilterType>("new");
+
+  // Add state for the sort order (default to 'desc' for descending)
+  const [sortOrder, setSortOrder] = useState("desc");
 
   const filters = ["completed", "new"];
```
Comment (xwdeng) [inline src/modclub_assets/src/components/app/profile/Activity.tsx:43]:
```suggestion
  const filters: FilterType[] = ["completed", "new"];
```

File: src/modclub_assets/src/components/app/profile/Activity.tsx [hunk 4]
```diff
@@ -42,13 +47,87 @@ export default function Activity() {
...
+          ? b.timestamp - a.timestamp
+          : a.timestamp - b.timestamp;
+      })
+      .map((item) => item.activity);
+  };
+
+  // Updates the activity list state
+  const updateActivitiesState = (activities, sortOrder) => {
+    const sortedActivities = sortActivities(activities, sortOrder);
+    //console.log("#2: ", currentFilter);
```
Comment (xwdeng) [inline src/modclub_assets/src/components/app/profile/Activity.tsx:89]:
```suggestion

```

File: src/modclub_assets/src/components/app/profile/Activity.tsx [hunk 5]
```diff
@@ -138,7 +246,7 @@ export default function Activity() {
       </div>
     </div>
   );
-
+  //console.log("#3:", inProgressActivity);
```
Comment (xwdeng) [inline src/modclub_assets/src/components/app/profile/Activity.tsx:249]:
```suggestion
  //console.log("#3:", inProgressActivity);
```

**Review by xwdeng — APPROVED** [review]

**Pull request merged**